ext/dba/dba.c

```diff
@@ -769,13 +771,17 @@ static void php_dba_open(INTERNAL_FUNCTION_PARAMETERS, bool persistent)
 			}
 		}
 		if (!info->lock.fp) {
+			zend_string *opened_path = NULL;
 			info->lock.fp = php_stream_open_wrapper(lock_name, lock_file_mode, STREAM_MUST_SEEK|REPORT_ERRORS|IGNORE_PATH|persistent_flag, &opened_path);
 			if (info->lock.fp) {
 				if (is_db_lock) {
+					ZEND_ASSERT(opened_path);
 					/* replace the path info with the real path of the opened file */
-					pefree(info->path, persistent);
-					info->path = pestrndup(ZSTR_VAL(opened_path), ZSTR_LEN(opened_path), persistent);
+					zend_string_release_ex(info->path, persistent);
```

Comment:
We don't need zend_string_release_ex() here, we can use zend_string_release(), as it knows whether the str is persistent or not